web/modules/custom/solution/solution.module

```diff
@@ -86,3 +88,25 @@ function solution_rdf_entity_insert(EntityInterface $entity) {
     $entity->set('field_is_affiliations_requests', NULL);
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ *
+ * Interrupts the permissions for updating a solution. The permission is not
+ * granted if there are no transitions allowed for the current state of the
+ * node.
+ */
+function solution_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if ($entity->bundle() != 'solution') {
+    return AccessResult::neutral();
+  }
+
+  if ($operation == 'edit') {
+    $allowed_transitions = $entity->get('field_is_state')->first()->getTransitions();
+    if (empty($allowed_transitions)) {
+      return AccessResult::forbidden();
```

Comment:
hook_entity_access is invoked when the Route check occurs. It shouldn't show the edit link if no permission is granted. Also, we already have tests for this. If this was shown, it would cause the features/solution/solution.moderation.feature to fail.